components/wifi/esp8266-driver/ESP8266/ESP8266.cpp

```diff
@@ -272,14 +271,15 @@ bool ESP8266::reset(void)
             continue;
         }
 
-        _rmutex.lock();
```

Comment:
Converting this as [PlatformMutex](https://os.mbed.com/docs/mbed-os/v5.14/apis/platformmutex.html) would be the right solution